sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogManager.scala

```diff
@@ -44,15 +44,35 @@ class CatalogManager(
   import CatalogManager.SESSION_CATALOG_NAME
 
   private val catalogs = mutable.HashMap.empty[String, CatalogPlugin]
+  // Map from catalog back to it's original name for easy name look up, we don't use the
+  // CatalogPlugin's name as it might be different from the catalog name depending on
+  // implementation. Catalog name <-> CatalogPlugin instance is a 1:1 mapping.
+  private val catalogIdentifiers = mutable.HashMap.empty[CatalogPlugin, String]
+  catalogIdentifiers(defaultSessionCatalog) = SESSION_CATALOG_NAME
```

Comment:
Maybe you should do this as part of `loadV2SessionCatalog`?

Reply:
No, the concern here is that if a user set a custom session catalog implementation and spark fail to load it, it would fall back to use `defaultSessionCatalog`. https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogManager.scala#L88.  So we need to set the defaultSessionCatalog in the reverse map here